terraform_env/dev/main.tf

```diff
@@ -0,0 +1,53 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 2.49"
+    }
+  }
+  backend "azurerm" {
+    resource_group_name  = "OpenCohort1_SultanShaikh_ProjectExercise"
+    storage_account_name = "tfstate007ss"
+    container_name       = "tfstate007ss"
+    key                  = "terraform_test.tfstate"
+  }
+}
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_cosmosdb_account" "main" {
+  name                = "${var.prefix}-cosmosdb-account"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  offer_type          = "Standard"
+  kind                = "MongoDB"
+  capabilities {
+    name = "EnableServerless"
+  }
+  capabilities {
+    name = "EnableMongo"
+  }
+  mongo_server_version = "3.6"
+
+  consistency_policy {
+    consistency_level       = "Session"
+    max_interval_in_seconds = 5
+    max_staleness_prefix    = 100
+  }
+
+  geo_location {
+    location          = var.location
+    failover_priority = 0
+  }
+
+  lifecycle { prevent_destroy = false }
+}
+
+data "azurerm_cosmosdb_account" "main" {
```

Comment:
Worth noting that we could take our test environment even further - currently you've switched your e2e test to run the app locally but talking to a test database, but we could have this terraform spin up the full test application and then run the test against the remote web address - i.e. the e2e test no longer needs to set up a web server at all, it just hits an existing web address